Sources/MongoSwift/BSON/Document+Codable.swift

```diff
@@ -1,219 +1,56 @@
-
-	// swiftlint:disable:next cyclomatic_complexity
-	private static func recursivelyEncodeKeyed(_ value: BsonValue?, forKey key: String, to container: inout KeyedEncodingContainer<_BsonKey>) throws {
-		let k = _BsonKey(stringValue: key)!
-		switch value {
-		case let val as Binary:
-			try container.encode(val, forKey: k)
-		case let val as Bool:
-			try container.encode(val, forKey: k)
-		case let val as Date:
-			try container.encode(val, forKey: k)
-		case let val as Decimal128:
-			try container.encode(val, forKey: k)
-		case let val as Double:
-			try container.encode(val, forKey: k)
-		case let val as Int:
-			try container.encode(val, forKey: k)
-		case let val as Int32:
-			try container.encode(val, forKey: k)
-		case let val as Int64:
-			try container.encode(val, forKey: k)
-		case let val as CodeWithScope:
-			try container.encode(val, forKey: k)
-		case let val as MaxKey:
-			try container.encode(val, forKey: k)
-		case let val as MinKey:
-			try container.encode(val, forKey: k)
-		case let val as ObjectId:
-			try container.encode(val, forKey: k)
-		case let val as String:
-			try container.encode(val, forKey: k)
-		case nil:
-			try container.encodeNil(forKey: k)
-		case let val as [BsonValue?]:
-			var nested = container.nestedUnkeyedContainer(forKey: k)
-			for v in val {
-				try Document.recursivelyEncodeUnkeyed(v, to: &nested)
-			}
-		case let val as Document:
-			var nested = container.nestedContainer(keyedBy: _BsonKey.self, forKey: k)
-			for (nestedK, nestedV) in val {
-				try Document.recursivelyEncodeKeyed(nestedV, forKey: nestedK, to: &nested)
-			}
-		default:
-			throw MongoError.typeError(message: "Encountered a non-encodable type in a Document: \(type(of: value))")
-		}
-	}
-
-	// swiftlint:disable:next cyclomatic_complexity
-	private static func recursivelyEncodeUnkeyed(_ value: BsonValue?, to container: inout UnkeyedEncodingContainer) throws {
-		switch value {
-		case let val as Binary:
-			try container.encode(val)
-		case let val as Bool:
-			try container.encode(val)
-		case let val as Date:
-			try container.encode(val)
-		case let val as Decimal128:
-			try container.encode(val)
-		case let val as Double:
-			try container.encode(val)
-		case let val as Int:
-			try container.encode(val)
-		case let val as Int32:
-			try container.encode(val)
-		case let val as Int64:
-			try container.encode(val)
-		case let val as CodeWithScope:
-			try container.encode(val)
-		case let val as MaxKey:
-			try container.encode(val)
-		case let val as MinKey:
-			try container.encode(val)
-		case let val as ObjectId:
-			try container.encode(val)
-		case let val as String:
-			try container.encode(val)
-		case nil:
-			 try container.encodeNil()
-		case let val as [BsonValue]:
-			var nested = container.nestedUnkeyedContainer()
-			for v in val {
-				try Document.recursivelyEncodeUnkeyed(v, to: &nested)
+			let key = _BsonKey(stringValue: k)!
+			if let val = v {
+				try container.encode(AnyBsonValue(val), forKey: key)
```

Comment:
Would it make sense to allow `AnyBsonValue` to be constructed with `nil`? That would simply this conditional.

Reply:
:+1: to making it feel like a `BsonValue`